security-utils/src/main/java/com/yahoo/security/tls/policy/PeerPolicy.java

```diff
@@ -48,12 +59,18 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) return false;
         PeerPolicy that = (PeerPolicy) o;
         return Objects.equals(policyName, that.policyName) &&
+                Objects.equals(description, that.description) &&
                 Objects.equals(assumedRoles, that.assumedRoles) &&
                 Objects.equals(requiredCredentials, that.requiredCredentials);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(policyName, assumedRoles, requiredCredentials);
+        return Objects.hash(policyName, description, assumedRoles, requiredCredentials);
+    }
+
+    @Override
+    public int compareTo(PeerPolicy o) {
+        return policyName.compareTo(o.policyName);
```

Comment:
Do we enforce policy name presence and/or uniqueness today? If not, what happens if someone adds multiple policies without setting names? Should this be a multi-dimensional comparison if this is an issue?

Reply:
Both are enforced in the constructor of `AuthorizedPeer`.
I just discovered that I have to rewrite `compareTo` to be consistent with `equals`. `TreeSet.equals` is currently reporting false positives 😨

Reply:
There is no simple way to implement `Comparable` to be "consistent with equals" (ref Javadoc for `Comparable`). I have removed the commit - it was really not necessary (only intended to improve assertion error messages in a related unit test).